sled-diagnostics/src/logs.rs

```diff
@@ -0,0 +1,245 @@
+    ZonePaths(anyhow::Error),
+}
+
+// Used for schemars to be able to be used with camino:
+// See https://github.com/camino-rs/camino/issues/91#issuecomment-2027908513
+fn path_schema(gen: &mut SchemaGenerator) -> Schema {
+    let mut schema: SchemaObject = <String>::json_schema(gen).into();
+    schema.format = Some("Utf8PathBuf".to_owned());
+    schema.into()
+}
+
+/// Provides all of the valid directores logs can be found in for this sled.
+pub fn valid_log_dirs() -> Result<BTreeSet<Utf8PathBuf>, SledDiagnosticsLogError>
+{
+    let zones =
+        oxlog::Zones::load().map_err(SledDiagnosticsLogError::ZonePaths)?;
+    Ok(zones
+        .zones
+        .into_values()
+        .flat_map(|entry| {
+            let mut paths = Vec::new();
+            paths.push(entry.primary);
+            paths.extend(entry.debug);
+            paths.extend(entry.extra.into_iter().map(|e| e.1));
+            paths
+        })
+        .collect())
+}
+
+/// Find the relevant logs that should be returned as apart of a Support Bundle.
+pub fn find_logs() -> Result<SledDiagnosticsLogs, SledDiagnosticsLogError> {
+    let zones =
+        oxlog::Zones::load().map_err(SledDiagnosticsLogError::ZonePaths)?;
+
+    let mut result = SledDiagnosticsLogs { logs: BTreeMap::new() };
+    for zone in zones.zones.keys() {
+        let filter = oxlog::Filter {
+            current: true,
+            archived: true,
+            extra: true,
+            // avoid having to call `lstat` on each log file
+            show_empty: true,
+        };
+
+        let mut logs = Service::new();
+        for (svc, mut svclogs) in zones.zone_logs(&zone, filter) {
+            // Current
+            if let Some(current) = svclogs.current {
+                logs.insert(svc.clone(), vec![current.path]);
+            }
+
+            // Archived
+            logs.entry(svc.clone()).or_default().extend(
+                svclogs.archived.into_iter().rev().take(5).map(|s| s.path),
+            );
+
+            // Extra
+            let extra = std::mem::take(&mut svclogs.extra);
+            for (_, cockroach) in process_extra_logs(extra) {
+                logs.entry(svc.clone()).or_default().extend(cockroach.current);
+                logs.entry(svc.clone())
+                    .or_default()
+                    .extend(cockroach.rotated.into_iter().rev().take(5));
+            }
+        }
+        result.logs.insert(zone.clone(), logs);
+    }
+
+    Ok(result)
+}
+
+#[derive(Default, Debug, PartialEq)]
+struct CockroachExtraLog {
+    current: Vec<Utf8PathBuf>,
+    rotated: Vec<Utf8PathBuf>,
+}
+
+/// Walk over the extra logs for cockroachdb preserving order and sort them into
+/// the appropriate buckets.
+fn process_extra_logs(
+    extra: Vec<oxlog::LogFile>,
+) -> HashMap<String, CockroachExtraLog> {
+    // Known logging paths for cockroachdb:
+    // https://www.cockroachlabs.com/docs/stable/logging-overview#logging-destinations
+    let cockroach_logs = [
```

Comment:
I know we discussed briefly on the weekly sync calls where this list of files should live but we never came to a conclusion. I am happy to move this to oxlog itself or to do so in a future PR when I end up shuffling some other oxlog internals around.